src/ptf/mask.py

```diff
@@ -63,36 +116,7 @@ def set_do_not_care_packet(self, hdr_type, field_name):
                 break
             else:
                 offset += bits
-        self.set_do_not_care(hdr_offset * 8 + offset, bitwidth)
-
-    def set_do_not_care_scapy(self, hdr_type, field_name):
-        warnings.warn(
-            '"set_do_not_care_scapy" is going to be deprecated, please '
-            'switch to the new one: "set_do_not_care_packet"',
-            DeprecationWarning,
-        )
-        self.set_do_not_care_packet(hdr_type, field_name)
-
-    def set_ignore_extra_bytes(self):
-        self.ignore_extra_bytes = True
-
-    def is_valid(self):
-        return self.valid
-
-    def pkt_match(self, pkt):
-        # just to be on the safe side
-        pkt = bytearray(bytes(pkt))
-        # we fail if we don't match on sizes, or if ignore_extra_bytes is set,
-        # fail if we have not received at least size bytes
-        if (not self.ignore_extra_bytes and len(pkt) != self.size) or len(
-            pkt
-        ) < self.size:
-            return False
-        exp_pkt = bytearray(bytes(self.exp_pkt))
-        for i in range(self.size):
-            if (exp_pkt[i] & self.mask[i]) != (pkt[i] & self.mask[i]):
-                return False
-        return True
+        return hdr_offset * 8 + offset, bitwidth
```

Comment:
Why remove the `set_do_not_care` call here? Shouldn't this be handled by the mask?

Reply:
I didn't remove the set_do_not_care call, this diff displaying is quite confused and incorrect. I will try to adjust the order of those functions to get a better diff display.

Reply:
Thanks for clarifying, the diff makes it difficult to understand what has been changed.

Reply:
I believe the confusion comes from adding `_calculate_fields_offset_and_bitwidth`. If you move that function above `set_do_not_care_packet` the diff should look better.

However, `set_do_not_care_packet` now calls into `_calculate_fields_offset_and_bitwidth` which does not use `self.set_do_not_care(hdr_offset * 8 + offset, bitwidth)`